### PR TITLE
feat: adding std::lib and refactor

### LINKED ISSUE #97: Allow reading (third party) libraries in the code for Miden Assembly
At the moment we create the Assembler with only `default()`

https://github.com/0xPolygonMiden/examples/blob/8b836be8a24d2bf95e5c0fc805198f7c17d6f612/playground/miden-wasm/src/lib.rs#L60

This way we are not able to allow users to compile programs using modules, see [here](https://wiki.polygon.technology/docs/miden/user_docs/stdlib/main#organization-and-usage). 

I think we need to read the code and extract the libraries that should be used. Then we instantiate the Assembler using the libraries.

### PR DESCRIPTION
**Pull request by Dominik1999**
Closes #97 and #99 

This PR includes the Miden StdLib. The program now compiles using an `Assembler` that uses 

```
        self.assembler = Assembler::default()
            .with_library(&StdLibrary::default())
            .map_err(|err| format!("Failed to load stdlib - {}", err))?
            .with_debug_mode(self.debug);
```

Furthermore, the Assembler can now also run in debug_mode.

I also refactored Rust Land a lot to introduce those changes. Now it should be prettier to read.

### CODE REVIEW

File: playground/miden-wasm/src/expected_test_proof.rs
Comment (Dominik1999) [inline playground/miden-wasm/src/expected_test_proof.rs:1]: This is just the proof and `cargo fmt` so you can skip this file in the review.

**Review by hackaugusto — APPROVED** [review]
I left a few comments, they are nits that would IMO improve the quality of the code, but I found no bugs and the playground behaves as expected, so I'm approving the PR

File: playground/miden-wasm/src/utils_debug.rs [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+        })
+    }
+
+    // MODIFIERS
+    // --------------------------------------------------------------------------------------------
+
+    /// executes a debug command against the vm in it's current state.
+    pub fn execute(&mut self, command: DebugCommand, param: Option<u64>) -> String {
+        match command {
+            //DebugCommand::PlayAll
```
[the remark this answers is not in the document]
Comment (Dominik1999) [reply playground/miden-wasm/src/utils_debug.rs:73]: very well

File: playground/miden-wasm/src/utils_debug.rs [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+            Some(prev_vm_state_result) => prev_vm_state_result.ok(),
+            None => None,
+        }
+    }
+
+    // ACCESSORS
+    // --------------------------------------------------------------------------------------------
+
+    /// print general VM state information.
+    pub fn print_vm_state(&self) -> String {
```
[the remark this answers is not in the document]
Comment (Dominik1999) [reply playground/miden-wasm/src/utils_debug.rs:151]: Will be removed in the next PR anyway so keeping this for now

File: playground/miden-wasm/src/utils_input.rs [hunk 3]
```diff
@@ -0,0 +1,117 @@
+use miden_vm::{
+    math::{Felt, FieldElement},
+    AdviceInputs, MemAdviceProvider, StackInputs, StackOutputs,
+};
+
+/// We derive Deserialize/Serialize so we can persist app state on shutdown.
```
[the remark this answers is not in the document]
Comment (Dominik1999) [reply playground/miden-wasm/src/utils_input.rs:6]: very good point.

File: playground/miden-wasm/src/utils_input.rs [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+
+                Ok(())
+            }
+        }
+    }
+
+    // Parse the outputs as str and return a vector of u64
+    pub fn deserialize_outputs(&mut self, outputs_as_str: &str) -> Result<(), String> {
+        let outputs_as_json: Outputs =
+            serde_json::from_str(&outputs_as_str).map_err(|e| e.to_string())?;
```
[the remark this answers is not in the document]
Comment (Dominik1999) [reply playground/miden-wasm/src/utils_input.rs:55]: I want the error as a String. I can't return the error to JS land. Only via the dev console in the browser. Will address error handling in https://github.com/0xPolygonMiden/examples/issues/98

File: playground/miden-wasm/src/utils_input.rs [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+            .unwrap_or(&[])
+            .iter()
+            .map(|v| v.parse::<u64>().map_err(|e| e.to_string()))
+            .collect::<Result<Vec<_>, _>>()?;
+
+        StackInputs::try_from_values(stack_inputs).map_err(|e| e.to_string())
+    }
+}
+
+//#[test]
```
[the remark this answers is not in the document]
Comment (Dominik1999) [reply playground/miden-wasm/src/utils_input.rs:102]: very well. I honestly like your sense for thoroughness.

File: playground/miden-wasm/src/lib.rs [hunk 6]
```diff
@@ -24,68 +15,24 @@ pub struct Outputs {
...
-    let program = assembler.compile(&asm).expect("Could not compile source");
-
-    let mut stack_input = StackInputs::new(vec![Felt::ZERO]);
-    let mut advice_provider = MemAdviceProvider::default();
-
-    if inputs_frontend.trim().is_empty() == false {
-        let inputs_des: InputFile = serde_json::from_str(&inputs_frontend)
-            .map_err(|err| format!("Failed to deserialize input data - {}", err))
-            .unwrap();
+    let mut program = utils_program::MidenProgram::new(code_frontend, false);
```
[the remark this answers is not in the document]
Comment (Dominik1999) [reply playground/miden-wasm/src/lib.rs:23]: good point

[the remark this answers is not in the document]
Comment (Dominik1999) [reply playground/miden-wasm/src/lib.rs:23]: That was more complicated than I thought. You cannot define enums as `bools` it seems https://stackoverflow.com/questions/52545734/is-there-a-way-to-use-a-boolean-as-an-underlying-type-for-an-enum

But now we have public constants

**Pull request merged**